CONFIGURATION.md

```diff
@@ -0,0 +1,297 @@
+### Proposal: No implicit defaults
+
+Therefore, a radical proposal: **we won't have any implicit defaults**. 
+At all. Strains won't magically inherit configuration values from:
+- the global properties in the `helix-config.yaml`
+- the command line
+- application (CLI) provided defaults
+- runtime (Fastly) provided defaults
+
+Unless you configure or reference a configuration value directly, it won't be effective. 
+Unless all required configuration values are set, you won't be able to deploy or publish.
+
+### Proposal: Generate a usable default configuration
+
+The flipside of this is that in order to enable development without any upfront configuration, we need to generate a usable default configuration taking the current context into consideration.
+
+A default configuration should be suggested when:
+- running `hlx deploy`
+- running `hlx publish`
+- running `hlx perf`
+- running `hlx demo`
+- outside of a CI environment
+
+A default configuartion should be saved when:
+- running `hlx * --save`
+- running `hlx up`
+
+The default configuration should have two strains:
+- `default` pointing to the git remote
+- `dev` pointing to localhost
+
+
+## Proposal: Move `url` under `condition`
+
+The `url` property for a strain is currently used as a shorthand for a strain matching condition (it also sets the base URL), so it would make sense to move it below `conditions`, especially considering that additional conditions may be added later.
+
+## Discussion: Parallel Deployments in a CI environment
+
+Concurrent deployments from a CI environment pose a hard problem at the moment:
+
+1. if the strain configuration cannot be modified by the deployment action, a new strain must be introduced for every new branch, prior to the creation of the branch, so that the branch can be deployed and tested
+2. if the strain configuration can be modified by the deployment action, a new strain can be introduced, but as the `helix-config.yaml` is the single point of synchronization, deployment in one branch will automatically deactivate all strains that refer to concurrent deployments made in different branches. In active development, this will lead to race conditions and intermittent test failures because the strain you just deployed got "undeployed" by a build triggered in another branch.
+3. even within one branch, making two committs in short succession will lead to the second commit tainting the deployment of the first build, with potentially misleading results. For instance a bug fix might get attributed to the first commit because this is where testing first succeeds, although the fix was deployed in the second commit
+
+
+### Proposal: Temporary Strains
+
+We introduce a new option for `hlx deploy --temp <tag[]?>` that takes zero or more tags for temporary deployments. By default, it uses `CIRCLE_SHA1` as the tag, i.e. the SHA1 hash of the last commit of the current build. This ties the deployment to the individual commit and avoids accidential overwriting.
```

Comment:
I think this section needs overhaul

Reply:
```suggestion
For testing a deployment in a Continuous Integration environment, it can be useful to have strains that are not persisted in the `helix-config.yaml`, but can still be activated for testing.
```